tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/integer_range_operators.bal

```diff
@@ -194,6 +194,86 @@ function testClosedIntRangeOnIntSubTypes() {
     assertEquality(arr[2], -4);
     assertEquality(arr[3], -3);
     assertEquality(arr[4], -2);
+
+    byte startValue11 = 7;
+    byte endValue11 = 10;
+    arr.removeAll();
+    foreach var val in startValue11 ... endValue11 {
+        arr.push(val);
+    }
+    assertEquality(arr.length(), 4);
+    assertEquality(arr[0], 7);
+    assertEquality(arr[1], 8);
+    assertEquality(arr[2], 9);
+    assertEquality(arr[3], 10);
+
+    int startValue12 = 7;
+    byte endValue12 = 10;
+    arr.removeAll();
+    foreach var val in startValue12 ... endValue12 {
+        arr.push(val);
+    }
+    assertEquality(arr.length(), 4);
+    assertEquality(arr[0], 7);
+    assertEquality(arr[1], 8);
+    assertEquality(arr[2], 9);
+    assertEquality(arr[3], 10);
+
+    int:Signed16|int:Unsigned32 startValue13 = 7;
+    int:Signed16|int:Unsigned32 endValue13 = 10;
+    arr.removeAll();
+    foreach var val in startValue13 ... endValue13 {
+        arr.push(val);
+    }
+    assertEquality(arr.length(), 4);
+    assertEquality(arr[0], 7);
+    assertEquality(arr[1], 8);
+    assertEquality(arr[2], 9);
+    assertEquality(arr[3], 10);
+
+    IntsV1 startValue14 = 4;
+    IntsV1 endValue14 = 6;
+    arr.removeAll();
+    foreach var val in startValue14 ... endValue14 {
+        arr.push(val);
+    }
+    assertEquality(arr.length(), 3);
+    assertEquality(arr[0], 4);
+    assertEquality(arr[1], 5);
+    assertEquality(arr[2], 6);
+
+    IntsV3 startValue15 = 1;
+    byte|IntsV2 endValue15 = 3;
+    arr.removeAll();
+    foreach var val in startValue15 ... endValue15 {
+        arr.push(val);
+    }
+    assertEquality(arr.length(), 3);
+    assertEquality(arr[0], 1);
+    assertEquality(arr[1], 2);
+    assertEquality(arr[2], 3);
+
+//    int:Signed32 startValue16 = +254;
+//    byte|IntsV3 endValue16 = 256;
```

Comment:
`byte|IntsV3 endValue16 = 256;` gives type checking error in the current master. Works in beta4. Will enable commented tests once the master is synced.